### PR TITLE
Send delivery receipt only to sender instead of everyone in conversation

### PR DESCRIPTION
**Pull request by marcoconti83**
# Reason for this pull request
Sending a delivery confirmation to all users of a conversation is a waste, only the sender needs to receive it.

# Changes
- If the message is a confirmation message, only the clients of the sender are included in the payload
- Moved around some test definitions for clearer structure
- Swiftification
- Now returning a `MissingClientsStrategy` to clarify whether the message should be sent only to one users or to all users in the conversation

# Testing
- Added testing for this feature

### CODE REVIEW

**Comment by marcoconti83** [comment]
BLOCKER: Waiting for testing this on staging backend, when we give green light it will be ported to production backend, then we can merge

**Comment by typfel** [comment]
@marcoconti83  How long will that take? Maybe could extract the fix for tests crashing into another PR?

**Comment by marcoconti83** [comment]
This PR completely changed, because of https://github.com/wireapp/wire-ios-data-model/pull/29/commits/a723180903dcfcce15732f9fcc63f313219c694f.

Please re-review.

File: Source/Model/Message/ZMAssetClientMessage+Encryption.swift [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        
+        var recipients : [ZMUserEntry] = []
+        selfClient.keysStore.encryptionContext.perform { (sessionsDirectory) in
+            recipients = genericMessage.recipientsWithEncryptedData(selfClient, recipients: conversation.activeParticipants.array as! [ZMUser], sessionDirectory: sessionsDirectory)
+        }
+        
+        builder.setRecipientsArray(recipients)
+        return builder.build()
+    }
+}
```
Comment (daehn) [inline Source/Model/Message/ZMAssetClientMessage+Encryption.swift]: Missing newline

**Comment by typfel** [comment]
So what changed? You also ported some code into swift...

**Comment by marcoconti83** [comment]
@typfel now the methods returning the encrypted data return a tuple, not just data, which specify what's the policy to adopt for missing clients.

File: Source/Model/Message/ZMAssetClientMessage+OTR.swift [hunk 2]
```diff
@@ -0,0 +1,9 @@
+//
+//  ZMAssetClientMessage+OTR.swift
+//  ZMCDataModel
+//
+//  Created by Marco Conti on 02/09/16.
```
Comment (daehn) [inline Source/Model/Message/ZMAssetClientMessage+OTR.swift]: Missing copyright header

File: Source/Model/Message/ZMClientMessage+Encryption.swift [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+/// us that some user clients that were supposed to be there are missing (e.g.
+/// another user added a new client that we don't yet know about). The various
+/// strategies give a hint to the backend of how we want to handle missing clients.
+public enum MissingClientsStrategy {
+    
+    /// Fail the request if there is any missing client
+    case DoNotIgnoreAnyMissingClient
+    /// Fail the request if there is any missing client for the given user,
+    /// but ignore missing clients of any other user
+    case IgnoreAllMissingClientsNotFromUser(user: ZMUser)
```
Comment (daehn) [inline Source/Model/Message/ZMClientMessage+Encryption.swift]: 👍

**Pull request merged**